AssertionError: 'b' unexpectedly found in ['a', 'b']

After fixing https://github.com/niwinz/django-redis/issues/366, I now [encounter on all Travis jobs](https://travis-ci.org/jayvdb/django-redis/builds/495996033). (And I also encounter this on my openSUSE builds, with [a workaround](https://build.opensuse.org/request/show/677659) )

```py
======================================================================
FAIL: test_keys (tests.test_backend.DjangoRedisCacheTestEscapePrefix)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/jayvdb/django-redis/tests/test_backend.py", line 104, in test_keys
    self.assertNotIn('b', keys)
AssertionError: 'b' unexpectedly found in ['a', 'b']

```

--- a/django_redis/client/sharded.py
+++ b/django_redis/client/sharded.py
@@ -220,7 +220,7 @@ class ShardClient(DefaultClient):
         raise NotImplementedError("iter_keys not supported on sharded client")
 
     def keys(self, search, version=None):
-        pattern = self.make_key(search, version=version)
+        pattern = self.make_pattern(search, version=version)
         keys = []
         try:
             for server, connection in self._serverdict.items():

--- a/tests/test_backend.py
+++ b/tests/test_backend.py
@@ -84,9 +84,6 @@ class DjangoRedisCacheTestEscapePrefix(unittest.TestCase):
         self.assertEqual(list(self.cache.iter_keys('*')), ['a'])
 
     def test_keys(self):
-        if isinstance(self.cache.client, ShardClient):
-            self.skipTest("ShardClient doesn't support iter_keys")
-
         self.cache.set('a', '1')
         self.other.set('b', '2')
         keys = self.cache.keys('*')

